catalog/core/catalog-core-migratable/src/test/java/org/codice/ddf/catalog/migratable/impl/MigrationTaskManagerTest.java

```diff
@@ -0,0 +1,253 @@
+        when(mockExecutor.awaitTermination(1L,
+                TimeUnit.MINUTES)).thenThrow(InterruptedException.class);
+
+        taskManager.exportFinish();
+
+        verify(mockExecutor).shutdown();
+        verify(mockExecutor).shutdownNow();
+    }
+
+    @Test
+    public void testExportFinishedTimeoutShutdown() throws Exception {
+        when(mockExecutor.awaitTermination(1L, TimeUnit.MINUTES)).thenReturn(false);
+
+        taskManager.exportFinish();
+
+        verify(mockExecutor).shutdown();
+        verify(mockExecutor).shutdownNow();
+    }
+
+    @Test(expected = MigrationException.class)
+    public void testExportFinishedErrorOccurred() throws Exception {
+        taskManager.getAtomicReference()
+                .set(mockThrowable);
+        taskManager.exportFinish();
+    }
+
+    @Test
+    public void testExportFinishedNoError() throws Exception {
+        taskManager.getAtomicReference()
+                .set(null);
+        taskManager.exportFinish();
+        //verify(mockAtomicRef).get(); // Can't execute this line - get() is final
+    }
+
+    @Test(expected = MigrationException.class)
+    public void testExportMetacardQueryWithError() throws Exception {
+        List<Result> testResults = new ArrayList<>();
+        for (int i = 0; i < TEST_RESULTS_LIST_SIZE; i++) {
+            testResults.add(new ResultImpl());
+        }
+
+        when(mockExecutor.submit(any(CatalogWriterCallable.class))).thenReturn(mockFutureInstance);
+
+        taskManager.getAtomicReference()
+                .set(mockThrowable);
+        taskManager.exportMetacardQuery(testResults, TEST_GROUP_COUNT);
+    }
+
+    @Test
+    public void testExportMetacardQuerySubmit() throws Exception {
+        List<Result> testResults = new ArrayList<>();
+        for (int i = 0; i < TEST_RESULTS_LIST_SIZE; i++) {
+            testResults.add(new ResultImpl());
+        }
+
+        config.setExportCardsPerFile(TEST_FILE_SIZE);
+
+        List<CatalogWriterCallable> writerCallableList = new ArrayList<>();
+
+        when(mockExecutor.submit(any(CatalogWriterCallable.class))).thenReturn(mockFutureInstance);
+
+        for (int i = 0; i < testResults.size(); i += config.getExportCardsPerFile()) {
+            File expectedFile = config.getExportPath()
+                    .resolve(taskManager.makeFileName(TEST_GROUP_COUNT, i))
+                    .toFile();
+            List<Result> expectedResults = testResults.subList(i,
+                    Math.min((i + config.getExportCardsPerFile()), testResults.size()));
+            CatalogWriterCallable writerCallable = new CatalogWriterCallable(expectedFile,
+                    expectedResults,
+                    mockFileWriter);
+            writerCallableList.add(writerCallable);
+        }
+
+        taskManager.exportMetacardQuery(testResults, TEST_GROUP_COUNT);
+
+        assertTrue(writerCallableList.equals(taskManager.getCallables()));
```

Comment:
We should use `assertThat(taskManager.getCallables(), containsInAnyOrder(writerCallableList.toArray()))` instead.